ClangPowerTools/ClangPowerTools/Helpers/SolutionInfo.cs

```diff
@@ -52,6 +54,36 @@ public static bool IsOpenFolderModeActive()
       return OpenFolderModeActive;
     }
 
+    public static bool ContainsCppProject()
+    {
+      DTE2 dte = (DTE2)VsServiceProvider.GetService(typeof(DTE));
+      Solution solution = dte.Solution;
+
+      if (solution == null)
+      {
+        return false;
+      }
+
+      return AnyCppProject(solution);
+    }
+
+    public static bool AnyCppProject(Solution solution)
+    {
+      foreach (var project in solution)
+      {
+        if (IsCppProject((Project)project))
+        {
+          return true;
+        }
+      }
+      return false;
+    }
+
+    public static bool IsCppProject(Project project)
```

Comment:
This method is not related to a `VS Solution`. Let's keep the `SolutionInfo` clean. Maybe a `ProjectInfo` class will be a good idea.